### ISSUE TITLE
Log `X-Pub-` headers

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/pub-dartlang-dart#74]
See https://github.com/dart-lang/pub/pull/1521

**Comment by mkustermann** [comment]
@kevmoo  Please specify exactly **what** data you would like to have associated with the headers and in what form the data should be saved!

Here is a suggestion of how we could do it:

* We identify a *pub request* by looking if there is at least one `x-pub-*` header.
* We insert one row into BigQuery for each *pub request*.
* We create a BigQuery table for each calendar day *yyyymmdd* and set an appropriate expiration date.
* We create the BigQuery tables with the following schema
```
timestamp      TIMESTAMP           The timestamp when the request was made.
host           STRING              The hostname used to contact the pub server.
path           STRING              The http path used to contact the pub server.
headers        RECORD    REPEATED  The X-Pub headers the client was sending.
headers.name   STRING              The X-Pub-* header name
headers.value  STRING              The X-Pub-* header value
```

This would be 
* forwards-compatible to new `x-pub-*` headers
* we avoid going through the logging api and bigquery sink
* we would avoid embedding the information in logging calls, intermixed with other logging messages 

wdyt?

**Comment by kevmoo** [comment]
Fixed in 337516cf0f7711f2531e916cf881a6576045ae94
